### PR TITLE
Show KV delete as async in reference docs

### LINKED ISSUE #625: KV Reference Docs don't indicate that `KV.delete()` is an async function.
There's no indication in either the example or in the documentation that deleting a key is an async operation. This could lead to race-conditions with other KV accesses in the same worker if not `await`ed upon or false positives with regard to successful deletes.

### PR DESCRIPTION
**Pull request by cwndrws**
Fixes #625

### CODE REVIEW

File: content/reference/storage/deleting-key-value-pairs.md [hunk 1]
```diff
@@ -12,6 +12,23 @@ This will remove the key and value from your namespace. As with any
 operations, it may take some time to see that they key has been deleted from
 various points at the edge.
 
+This method returns a promise that you should `await` on in order to verify
+successful deletion.
+
+Here's an example of deleting a key from within a Worker:
+
+```js
+addEventListener('fetch', event => {
```
Comment (victoriabernard92) [inline content/reference/storage/deleting-key-value-pairs.md]: I would leave out the script context and just say `await KV_NAMESPACE.delete("some-key")`

**Comment by victoriabernard92** [comment]
Closing in favor of https://github.com/cloudflare/workers-docs/pull/628

Note in redesign it will be much clearer the args and return types of runtime APIs

**Comment by victoriabernard92** [comment]
Reopening this per Steve's point. I suggest we:
* Clarify the top example which is likely where the users will copy-paste from (#628 ) 
* Add a short note that the await is required to perform the delete

**Comment by cwndrws** [comment]
@victoriabernard92 good call. I updated the top example and added a note. Let me know if anything else needs to change.

**Pull request merged**